src/types/pgduckdb_enum.cpp

```diff
@@ -0,0 +1,140 @@
+#include "pgduckdb/types/pgduckdb_enum.hpp"
+#include "pgduckdb/pgduckdb_utils.hpp"
+#include "duckdb/common/extra_type_info.hpp"
+
+namespace pgduckdb {
+
+using duckdb::idx_t;
+using duckdb::LogicalTypeId;
+using duckdb::PhysicalType;
+
+// This is medium hacky, we create extra space in the Vector that would normally only hold the strings corresponding to
+// the enum's members. In that extra space we store the enum member oids.
+
+// We do this so we can find the OID from the offset (which is what DuckDB stores in a Vector of type ENUM) and return
+// that when converting the result from DuckDB -> Postgres
```

Comment:
This is not true anymore right? i.e. you switched back to the first with the vendored in template (until the next duckdb release)